### PR TITLE
Add a warning if patch doesn't succeed

### PR DESCRIPTION
**Pull request by michaelnarodovitch**
## What do these changes do?

The situation
- kopf submits a `merge-patch+json` patch
- kubernetes API doesn't apply the `merge-patch+json` patch but responds `200 OK`

leads to bugs and confusion, when testing operators.

With this change, kopf will check, if the `merge-patch+json` patch was merged and be verbose, if the patch was not exactly successful.

## Description

**Before:** When a patch wasn't merged, then kopf didn't report anything.

**After:** When a patch wasn't merged, then kopf will report if ran in `--verbose` mode.

This will help users to debug their operators. Especially, when they use OpenAPI schemas in their CRDs.

Not sure if the check is in the right place and the logger is setup appropriately.

No risky changes.

## Issues/PRs

 

If a patch is not merged, then it is usually because of a schema mismatch of the CRD. For this reason it was discussed to move the internal kopf `status` to the annotations.  #308 #321

## Type of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] The code addresses only the mentioned problem, and this problem only
- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`

### CODE REVIEW

File: kopf/clients/patching.py [hunk 1]
```diff
@@ -1,13 +1,18 @@
-from typing import Optional, cast
+import copy
+from typing import Optional, cast, Any, Union
 
 import aiohttp
 
 from kopf.clients import auth
 from kopf.clients import discovery
-from kopf.structs import bodies
+from kopf.structs import bodies, diffs
```
Comment (nolar) [inline kopf/clients/patching.py:8]: It is better to keep the code style consistent with the surroundings. I regularly (once a week or two) think on switching to this PyCharm-suggested form of one-line imports (as it does on every refactoring or move moves), but this should be done as a separate PR for all the modules — for consistency.

Meanwhile, let's follow the existing [Google Style Guide for imports](http://google.github.io/styleguide/pyguide.html#313-imports-formatting).

**Comment by michaelnarodovitch** [comment]
Thx for all the comments and suggestions. Had to add quite a bit of code. I think quality is OK and changes are safe.

Missing is integration tests with the Kubernetes API responses, and especially integration tests with the `status_patch` merging. What I did was based of an educated guess from reviewing https://github.com/kubernetes/apiextensions-apiserver. If we wanted to be 100% safe, we would also have to test the `PATCH` API of the canonical Kubernetes objects.

File: kopf/clients/patching.py (hunk 1 above)
Comment (michaelnarodovitch) [reply kopf/clients/patching.py]: Actually, I like the google style more as it produces cleaner diffs (one line per unit change). Can be checked automatically with https://github.com/PyCQA/flake8-import-order . Pycharm can be configured to do google style imports https://www.jetbrains.com/help/pycharm/configuring-code-style.html.

**Comment by nolar** [comment]
Implemented in https://github.com/nolar/kopf/pull/527

**Pull request closed without merging**